### PR TITLE
improvement: S3C-3316 modify notification message

### PR DESCRIPTION
**Pull request by naren-rajendran**
Bucket notification message structure is modified to match aws specifications.

### CODE REVIEW

**Review by dora-korpar — APPROVED** [review]

File: extensions/notification/utils/message.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * @param  {Object} message - initial message generated
+ * @return {Object} updated message object
+ */
+function addLogAttributes(logEntry, message) {
+    const { eventMessageProperty } = constants;
+    const msg = {};
+    for (const [key, value] of Object.entries(eventMessageProperty)) {
+        let val = logEntry[value];
+        if (typeof val === 'number') {
+            val = val.toString();
```
Comment (jonathan-gramain) [inline extensions/notification/utils/message.js:19]: Is this required by the spec (convert numbers to strings)?

File: extensions/notification/utils/message.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                },
+                requestParameters: {
+                    sourceIPAddress: null,
+                },
+                responseElements: {
+                    'x-amz-request-id': null,
+                    'x-amz-id-2': null,
+                },
+                s3: {
+                    s3SchemaVersion: event.schemaVersion,
```
Comment (jonathan-gramain) [inline extensions/notification/utils/message.js]: Why is the schema version part of the `event` object? I imagined (but can be wrong) that the schema version is a AWS API version like `2006-07-01`

File: extensions/notification/utils/message.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    return Object.assign(msg, message);
+}
+
+/**
+ * Transforms an event object to specification.
+ * @param  {Object} event - event object
+ * @return {Object} updated event object
+ */
+function transformToSpec(event) {
+    const verId
```
Comment (jonathan-gramain) [inline extensions/notification/utils/message.js]: You may call this var `encodedVersionId` for clarity

File: extensions/notification/utils/message.js (hunk 1 above)
Comment (naren-rajendran) [reply extensions/notification/utils/message.js:19]: yes everything appears to be string, https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html

File: extensions/notification/utils/message.js (hunk 2 above)
Comment (naren-rajendran) [reply extensions/notification/utils/message.js]: I wasn't sure and the doc I referred wasn't clear either, https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html

Comment (jonathan-gramain) [reply extensions/notification/utils/message.js]: Ok, maybe it's  better to have it defined as a constant `1.0` then, if it refers to the version of this `message` object it should probably not be part of the `event`.

**Review by jonathan-gramain — APPROVED** [review]

File: extensions/notification/utils/message.js (hunk 2 above)
Comment (naren-rajendran) [reply extensions/notification/utils/message.js]: @jonathan-gramain there is another property `eventVersion` in the message, this probably is  the one that relates to version of message object. Should I still make the `s3SchemaVersion` to be `1.0`?

Comment (jonathan-gramain) [reply extensions/notification/utils/message.js]: In doubt, I'd set `s3SchemaVersion` to `1.0` since this is what is shown as an example in the AWS page.

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e approve

**Comment by naren-scality** [comment]
@bert-e resest

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e reset

**Pull request merged**